buildSrc/src/main/kotlin/JpackageExec.kt

```diff
@@ -0,0 +1,313 @@
+     */
+    @get:Input
+    @get:Optional
+    val winUpgradeUuid = stringProperty()
+
+    /**
+     * Windows-specific. The name to use for the application in its Windows registry entries.
+     */
+    @get:Input
+    @get:Optional
+    val winRegistryName = stringProperty()
+
+    /**
+     * Windows-specific. Allows the application to be installed in the Windows start menu.
+     */
+    @get:Input
+    @get:Optional
+    val addToWindowsMenu = objectFactory.property<Boolean>()
+
+    /**
+     * Windows-specific. Adds a shortcut to the user's desktop.
+     */
+    @get:Input
+    @get:Optional
+    val addWindowsDesktopShortcut = objectFactory.property<Boolean>()
+
+    /**
+     * Mac-specific. An identifier to use for the App Store.
+     */
+    @get:Input
+    @get:Optional
+    val macBundleIdentifier = stringProperty()
+
+    @TaskAction
+    fun exec() {
+        project.exec {
+            val args = mutableListOf<String>()
+            args.add(jdkHome.file("bin/jpackage").get().asFile.absolutePath)
+            args.add("create-installer")
+
+            runtimeImage.ifPresent { dir ->
+                args.addAll("--runtime-image", dir.asFile.absolutePath)
+            }
+            if (verbose.getOrElse(false)) {
+                args.add("--verbose")
+            }
+            jvmArgs.ifPresent { jvmArgs ->
+                args.add("--jvm-args")
+                args.add(jvmArgs.joinToString(separator = " ", prefix = "\"", postfix = "\""))
+            }
+            args.addAll("--input", inputDir.get().asFile.absolutePath)
+            resourceDir.ifPresent { dir ->
+                args.addAll("--resource-dir", dir.asFile.absolutePath)
+            }
+            args.addAll("--output", outputDir.get().asFile.absolutePath)
+            icon.ifPresent { iconFile ->
+                args.addAll("--icon", iconFile.toString())
+            }
+            args.addAll("--main-jar", mainJar.get().asFile.absolutePath)
+            args.addAll("--main-class", mainClassName.get())
+
+            args.addAll("--name", applicationName.get())
+            applicationDescription.ifPresent { description ->
+                args.addAll("--description", description)
+            }
+            args.addAll("--app-version", applicationVersion.get())
+            copyright.ifPresent {
+                args.addAll("--copyright", it)
+            }
+            licenseFile.ifPresent { file ->
+                args.addAll("--license-file", file.asFile.absolutePath)
+            }
+            applicationVendor.ifPresent { vendor ->
+                args.addAll("--vendor", vendor)
+            }
+            identifier.ifPresent { id ->
+                args.addAll("--identifier", id)
+            }
+            fileAssociations.ifPresent { propsFile ->
+                args.addAll("--file-associations", propsFile.asFile.absolutePath)
+            }
+            args.addAll("--installer-type", installerType.get())
```

Comment:
Consider extracting all of this to a private function?

Reply:
All of what?

Reply:
The arg creation.